pkg/controller/certificaterequests/controller.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificaterequests
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/utils/clock"
+
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+	cmlisters "github.com/jetstack/cert-manager/pkg/client/listers/certmanager/v1alpha1"
+	controllerpkg "github.com/jetstack/cert-manager/pkg/controller"
+	"github.com/jetstack/cert-manager/pkg/issuer"
+	logf "github.com/jetstack/cert-manager/pkg/logs"
+	"github.com/jetstack/cert-manager/pkg/metrics"
+)
+
+const (
+	ControllerName = "certificaterequests"
+)
+
+var keyFunc = controllerpkg.KeyFunc
+
+type Controller struct {
+	helper issuer.Helper
+
+	// clientset used to update cert-manager API resources
+	cmClient cmclient.Interface
+
+	certificateRequestLister cmlisters.CertificateRequestLister
+
+	queue   workqueue.RateLimitingInterface
+	metrics *metrics.Metrics
+
+	// used for testing
+	clock clock.Clock
+
+	// logger to be used by this controller
+	log logr.Logger
+
+	// used to record Events about resources to the API
+	recorder record.EventRecorder
+
+	// the issuer kind to react to when a certificate request is synced
+	issuerType string
+
+	issuerLister        cmlisters.IssuerLister
+	clusterIssuerLister cmlisters.ClusterIssuerLister
+	issuerFactory       issuer.IssuerFactory
+}
+
+func New(issuerType string) *Controller {
+	return &Controller{
+		issuerType: issuerType,
```

Comment:
Does this need to be a field on the struct, or can it be an unexported const?

Reply:
How would that work unless all code referencing it moves to `./pkg/controller/certificiaterequest/$issuer/.`?